File: ios/model/PlantMetadata.swift
import Foundation

struct PlantMetadata: Serializable {
    let uuid: String
    let title: String
    let subtitle: String
    let timestamp: Date
    let image: String
    static func toObject(map: [String: Any]) -> PlantMetadata? {
        return map["uuid"] == nil || map["title"] == nil || map["subtitle"] == nil || map["timestamp"] == nil || map["image"] == nil ? nil : PlantMetadata(uuid: map["uuid"]! as! String, title: map["title"]! as! String, subtitle: map["subtitle"]! as! String, timestamp: DateService.toDate(from: map["timestamp"]! as! String), image: map["image"]! as! String)
    }
}

File: android/model/PlantMetadata.kt
package com.tobiasheymann.ep.model

import java.io.Serializable
import java.time.Instant

data class PlantMetadata(
    val uuid: String,
    val title: String,
    val subtitle: String,
    val timestamp: Instant,
    val image: String
) : Serializable {
    companion object {
        fun toObject(map: Map<String, Any>): PlantMetadata? {
            return if (
                map["uuid"] == null
                || map["title"] == null
                || map["subtitle"] == null
                || map["timestamp"] == null
                || map["image"] == null
            )
                null
            else
                PlantMetadata(
                    map["uuid"]!! as String,
                    map["title"]!! as String,
                    map["subtitle"]!! as String,
                    Instant.parse(map["timestamp"]!! as String),
                    map["image"]!! as String
                )
        }
    }
}
